src/ImageRunModal.jsx

```diff
@@ -633,6 +588,82 @@ export class ImageRunModal extends React.Component {
+        return Object.keys(validationFailed).length == 0;
+    };
+
+    async validateContainerName(containerName) {
+        try {
+            await client.containerExists(this.isSystem(), containerName);
+        } catch (error) {
+            return;
+        }
+        return _("Name already in use");
+    }
+
+    async validateForm() {
+        const { publish, volumes, env, containerName } = this.state;
+        const validationFailed = { };
+
+        const publishValidation = publish.map(a => {
+            let publishPortValidation;
+            publishPortValidation = validatePublishPort(a.IP, "IP", publishPortValidation);
+            publishPortValidation = validatePublishPort(a.hostPort, "hostPort", publishPortValidation);
+            publishPortValidation = validatePublishPort(a.containerPort, "containerPort", publishPortValidation);
+
+            return publishPortValidation;
+        });
+        if (publishValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.publish = publishValidation;
+
+        const volumesValidation = volumes.map(a => {
+            let volumesValidation;
+            volumesValidation = validateVolume(a.hostPath, "hostPath", volumesValidation);
+            volumesValidation = validateVolume(a.containerPath, "containerPath", volumesValidation);
+
+            return volumesValidation;
+        });
+        if (volumesValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.volumes = volumesValidation;
+
+        const envValidation = env.map(a => {
+            let envValidation;
+            envValidation = validateEnvVar(a.envKey, "envKey", envValidation);
+            envValidation = validateEnvVar(a.envValue, "envValue", envValidation);
+
+            return envValidation;
+        });
+        if (envValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.env = envValidation;
+
+        const containerNameValidation = await this.validateContainerName(containerName);
+
+        if (containerNameValidation)
+            validationFailed.containerName = containerNameValidation;
+
+        this.setState({ validationFailed });
+
+        return this.isFormValid(validationFailed);
+    }
+
+    /* Updates a validation object of the whole dynamic list's form (e.g. the whole port-mapping form)
+    *
+    * Arguments
+    *   - key: [publish/volumes/env] - Specifies the validation of which dynamic form of the Image run dialog is being updated
+    *   - value: An array of validation errors of the form. Each item of the array represents a row of the dynamic list.
+    *            Index needs to corellate with a row number
+    */
+    dynamicListOnValidationChange = (value, key) => {
+        const validationFailedDelta = { ...this.state.validationFailed };
+
+        validationFailedDelta[key] = value;
+
+        if (validationFailedDelta[key].every(a => a === undefined))
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231009-115441-607719f9-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)